Microsoft.NET.Build.Containers/Layer.cs

```diff
@@ -26,21 +27,26 @@ public static Layer FromFiles(IEnumerable<(string path, string containerPath)> f
     {
         long fileSize;
         Span<byte> hash = stackalloc byte[SHA256.HashSizeInBytes];
+        byte[] uncompressedHash;
 
         string tempTarballPath = ContentStore.GetTempFile();
         using (FileStream fs = File.Create(tempTarballPath))
         {
-            // using (GZipStream gz = new(fs, CompressionMode.Compress)) // TODO: https://github.com/dotnet/sdk-container-builds/issues/29
-            using (TarWriter writer = new(fs, TarEntryFormat.Gnu, leaveOpen: true))
+            using (HashDigestGZipStream gz = new(fs, leaveOpen: true))
             {
-                foreach (var item in fileList)
+                using (TarWriter writer = new(gz, TarEntryFormat.Gnu, leaveOpen: true))
                 {
-                    // Docker treats a COPY instruction that copies to a path like `/app` by
-                    // including `app/` as a directory, with no leading slash. Emulate that here.
-                    string containerPath = item.containerPath.TrimStart(PathSeparators);
+                    foreach (var item in fileList)
+                    {
+                        // Docker treats a COPY instruction that copies to a path like `/app` by
+                        // including `app/` as a directory, with no leading slash. Emulate that here.
```

Comment:
@rainersigwald I know this was before the PR, but I'm not understanding this comment clearly: Is `/app` being prefixed or suffixed to the file path?

I ask because if it's a prefix, and it needs to be removed, we could try using the stream-based `TarFile.CreateFromDirectory` with `includeBaseDirectory=false`, see if that works.

If that's not what the comment meant, then nothing to do here.